### PR TITLE
Update docs for Nerves 0.5.0 / Bootstrap 0.3.0 release

### PR DESCRIPTION
**Pull request by GregMefford**
Prepping documentation for release of `nerves` 0.5.0 and `nerves_bootstrap` 0.3.0.

- [x] Getting Started page
- [x] Installation page
- [x] Systems page
- [x] Targets page
- [x] User Interfaces page
- [x] Advanced Configuration page
- [x] FAQ page
- [x] How to use the Docker provider
  - [x] in Installation
  - [x] in Systems

### CODE REVIEW

File: docs/Getting Started.md [hunk 1]
```diff
@@ -28,118 +30,100 @@ firmware image | Built from a firmware bundle and contains the partition table,
...
-
-You'll notice that subsequent calls to `compile` will not fetch or build the system because they're already cached on your host computer.
 
-## Making Firmware
+## Building and Deploying Firmware
 
-Now that you have a compiled Nerves application, you can produce firmware. Nerves firmware is the product of turning your application into an OTP release, adding it to the system image, and laying out a partition scheme. You can create the firmware bundle with the following command:
+Once the dependencies are fetched, you can build a Nerves Firmware (a bundle that contains a minimal Linux platform and your application, packaged as an OTP release).
+The first time you ask any dependencies or your application to compile, Nerves will fetch the System and Toolchain from one of our cache mirrors.
+These Artifacts are cached locally in `~/.nerves/artifacts` so they can be shared across projects.
```
Comment (mobileoverlord) [inline docs/Getting Started.md]: Artifacts should probably be lowercased

File: docs/Installation.md [hunk 2]
```diff
@@ -1,66 +1,93 @@
...
+Optionally, if you want to build custom Nerves Systems, you'll also need to install [Docker for Mac](https://www.docker.com/products/overview#/install_the_platform).
+
 Now skip to the instructions for all platforms below.
 
 ## Linux
 
-We've found quite a bit of variation between Linux distributions. Nerves requires Erlang versions `>= 19.0` and Elixir versions `>= 1.2.4`.
-If you need to install Erlang, see the prebuilt versions and guides provided by [Erlang Solutions](https://www.erlang-solutions.com/resources/download.html)
+We've found quite a bit of variation between Linux distributions.
+Nerves requires Erlang versions `>= 19.0` and Elixir versions `>= 1.2.4`.
```
Comment (mobileoverlord) [inline docs/Installation.md]: I think we landed on stating >= 1.4.0 due to moves to registry

File: docs/FAQ.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ This is a collection of questions that often come up as people are getting start
 If you tried to go through the [Getting Started guide](https://hexdocs.pm/nerves/getting-started.html) or some of the [example projects](https://github.com/nerves-project/nerves-examples) and got stuck, hopefully one of the following answers will help.
 If not, please let us know in the #nerves channel on [the Elixir-Lang Slack](https://elixir-slackin.herokuapp.com/), or [create an Issue or Pull Request to improve this documentation](https://github.com/nerves-project/nerves/tree/master/docs).
 
-### How Do I Connect to the Target using a Serial Cable Instead of Using the HDMI Screen?
+## Using a USB Serial Console
 
 By default, the `iex` console is displayed on the screen attached to the HDMI port, which tends to be easier for new people because they can simply connect their target device to a monitor or TV.
```
Comment (mobileoverlord) [inline docs/FAQ.md]: I believe that "iex console is displayed on the screen attached to the HDMI port" is only true for `rpi*` targets

**Comment by GregMefford** [comment]
This last commit also resolves #87 and #88.

**Pull request merged**